### ISSUE TITLE
Allow ignoring errors when deserializing null value into primitives

### ISSUE THREAD

**Issue by moghaddam** [eclipse-ee4j/jsonb-api#215]
In Jackson, developers can ignore null values in deserialization of the primitive types by setting the  `FAIL_ON_NULL_FOR_PRIMITIVES` configuration to false. Recently I've been asked by a couple of developers who were migrating their code to json-b, about our solution to support this issue. Unfortunately there is no way to achieve the same result in json-b (except maybe by providing deserializers for all primitive types that's totally non-sense).
Is this something that is discussed previously and is not in jsonb due to technical/design issues? If the anser is no, do you think we can consider it as an option for upcoming versions?

**Comment by aguibert** [comment]
hi @moghaddam thanks for raising this issue. This seems like a reasonable request. Currently the spec says the following on this topic:

> The deserialization operation of a property with a null value in a JSON document MUST set the value of the field to null value (or call setter with null value if setter is present). The exception is java.util.Optional, OptionalInt, OptionalLong, OptionalDouble instances. In this case the value of the field is set to an empty optional value.

I am curious, do you know why these values are becoming null in the first place? Is the client sending the JSON data from a non-Java language?

**Comment by rmannibucau** [comment]
Shouldnt it default to the primitive default? Failling sounds like a bug since wrappers enable to handle the absence of value.

**Comment by aguibert** [comment]
Yes, I agree the current behavior could be improved. I think we should change the spec so that null values for primitives are ignored by JSON-B (i.e. don't set the field or call the setter with the default primitive value)

**Comment by moghaddam** [comment]
Hi Andrew

Yes, in my example serialization was not done in json-b.
